picker/leastloaded.go

```diff
@@ -154,7 +155,21 @@ func (h *leastLoadedConnHeap) update(allConns conn.Conns) {
 		}
 	}
 	newLen := j + len(newMap)
-	if len(slice) > newLen {
+	if j == len(slice) {
+		// No items removed, so we haven't broken any heap invariants.
+		// If we don't have too many items to add, just heap.Push them
+		// and return.
+		threshold := newLen / bits.Len(uint(newLen))
```

Comment:
We could possibly avoid this computation, too, with:
```suggestion
		if len(newMap) == 0 {
			return
		}
		threshold := newLen / bits.Len(uint(newLen))
```
Thoughts?

FWIW, I haven't done any benchmarks. This code change is just from first principles, so maybe not worth adding (without actually benchmarking 🤷).

Reply:
Seems fine to me. I mean, as far as I can tell, it's trivially correct anyway. I don't think it makes this much worse.

Reply:
Thinking about this more, this is probably overkill.

Even if there were 1000s of backends and no subsetting, we're still only talking about infrequent (only when the resolver updates) added latency of a few microseconds, for RPCs waiting for this to complete so they can acquire the lock. I suspect lock contention would generally be a much bigger issue for throughput than this proposed micro-optimization, especially since we can't use a reader-writer lock -- all RPCs must acquire and release the write lock.

¯\\_(ツ)\_/¯